contrib/linux-kernel/lib/zstd/compress.c

```diff
@@ -587,8 +587,10 @@ ZSTD_STATIC size_t ZSTD_compressSequences(ZSTD_CCtx *zc, void *dst, size_t dstCa
 {
 	const int longOffsets = zc->params.cParams.windowLog > STREAM_ACCUMULATOR_MIN;
 	const seqStore_t *seqStorePtr = &(zc->seqStore);
-	U32 count[MaxSeq + 1];
-	S16 norm[MaxSeq + 1];
+	U32 *count = (U32 *)zc->tmpCounters;
+	S16 *norm = (S16 *)(count + MaxSeq + 1);
```

Comment:
`-Wcast-align` is supposed to trigger a warning if one upgrade a `U16` pointer into a `U32` one,
even if you manually take good care of address alignment,
which is what happens below when `norm` get used as a base for `workspace`.

The order in memory can be different.
You can keep `norm` right after `count` if it's better,
but `workspace` will have to be based from `count`, because it is also a `U32`.